packages/base-zone/src/prepare-revocable.js

```diff
@@ -153,3 +185,83 @@ export const prepareRevocableMakerKit = (
   });
 };
 harden(prepareRevocableMakerKit);
+
+/**
+ * Create an exo class kit used to guard operations and cancel them.
+ *
+ * @param {import('@agoric/base-zone').Zone} zone
+ * @param {(canceller: {
+ *   cancel: () => boolean;
+ *   revoke: () => boolean; // Alias for cancel (for revocable objects).
+ *   release: () => void;
+ * }) => void} [onRelease]
+ *   Clean up any additional resources for the guarded object.
+ */
+export const prepareCancelKit = (zone, onRelease = () => {}) => {
+  const makeCancelKit = zone.exoClassKit(
+    'CancelKit',
+    {
+      canceller: M.interface('Canceller', {
+        revoke: M.call().returns(M.boolean()),
+        cancel: M.call().returns(M.boolean()),
+        release: M.call().returns(),
+      }),
+      guard: M.interface('CancellationGuard', {
+        assert: M.call().returns(),
+        isCancelled: M.call().returns(M.boolean()),
+        reason: M.call().returns(M.or(M.error(), M.undefined())),
+      }),
+    },
+    /**
+     * @type {(label: string) => { cancelled: boolean; label: string }}
+     */
+    label => ({ cancelled: false, label }),
+    {
+      canceller: {
+        cancel() {
+          const { cancelled } = this.state;
+          if (cancelled) {
+            return false;
+          }
+          this.state.cancelled = true;
+          this.facets.canceller.release();
+          return true;
+        },
+        revoke() {
+          // Alias for cancel, for revocable objects.
+          return this.facets.canceller.cancel();
+        },
+        release() {
+          onRelease(this.facets.canceller);
+        },
+      },
+      guard: {
+        assert() {
+          const {
+            facets: { guard },
+          } = this;
```

Comment:
in the interest of conciseness:

```suggestion
          const { guard } = this.facets;
```

it's just a personal preference. It bugs me when the linter adds linebreaks for no good reason, reducing the amount of code I can see on a page.

Reply:
Removed this code entirely.